addon/mixins/model-records/edit-actions-mixin.js

```diff
@@ -0,0 +1,47 @@
+import Ember from 'ember'
+
+const {
+  Mixin,
+  get,
+  RSVP
+} = Ember
+
+export default Mixin.create({
+  actions: {
+    save() {
+      return get(this, 'model')
+        .save()
+        .then(record => {
+          this.transitionToRoute(
+            'model-records',
+            record.constructor.modelName
```

Comment:
For anyone reviewing who, like me, isn't familiar with the Javascript prototype class system, `.constructor` returns a function that is analogous to a "Class".

```js
$E.constructor.modelName // "user"
'hi'.constructor // function String() { [native code] }
```